### PR TITLE
Adding access_token validation for RS256 id_token's

### PR DESCRIPTION
**Pull request by luisrudge**

### CODE REVIEW

File: src/helper/error.js [hunk 1]
```diff
@@ -5,11 +5,11 @@ function buildResponse(error, description) {
   };
 }
 
-function invalidJwt(description) {
+function invalidToken(description) {
```
Comment (lbalmaceda) [inline src/helper/error.js]: can this be considered a breaking change?

File: test/web-auth/web-auth.test.js [hunk 2]
```diff
@@ -307,6 +317,75 @@ describe('auth0.WebAuth', function() {
         }
       ); // eslint-disable-line
     });
+    it('should return the id_token payload when there is no access_token', function(done) {
```
Comment (lbalmaceda) [inline test/web-auth/web-auth.test.js]: shouldn't the payload also be returned when there's a VALID access token? That's the missing test case IMO, even if it's 95% similar to this

File: test/web-auth/web-auth.test.js [hunk 3]
```diff
@@ -307,6 +317,120 @@ describe('auth0.WebAuth', function() {
...
+            state: 'foo',
+            expiresIn: null,
+            tokenType: 'Bearer',
+            scope: null
+          });
+          done();
+        }
+      ); // eslint-disable-line
+    });
+    it('should return the id_token payload when there is an access_token', function(done) {
```
Comment (lbalmaceda) [inline test/web-auth/web-auth.test.js]: when there is a _valid_ access_token

File: test/web-auth/web-auth.test.js [hunk 4]
```diff
@@ -307,6 +317,120 @@ describe('auth0.WebAuth', function() {
...
+            state: 'foo',
+            expiresIn: null,
+            tokenType: 'Bearer',
+            scope: null
+          });
+          done();
+        }
+      ); // eslint-disable-line
+    });
+    it('should validate an access_token when available', function(done) {
```
Comment (lbalmaceda) [inline test/web-auth/web-auth.test.js]: I don't know if you are using the same convention somewhere, but I'd say:

```
should throw a token validation error when the access_token is available but not valid
```

Comment (luisrudge) [reply test/web-auth/web-auth.test.js]: image

**Pull request merged**